### PR TITLE
Fix PHP invalid index warning

### PR DESCRIPTION
**Pull request by splitice**
Fix PHP Notice: Undefined offset: -1 in getLastError

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please follow PSR-2.

**Comment by GrahamCampbell** [comment]
Also, what version of php>

**Comment by splitice** [comment]
Every version of PHP since PHP 4 has had a warning for invalid index offset.

File: phpseclib/Net/SSH2.php [hunk 1]
```diff
@@ -3774,7 +3774,11 @@ function getErrors()
      */
     function getLastError()
     {
-        return $this->errors[count($this->errors) - 1];
+        $count = count($this->errors);
+        if($count == 0) {
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: missing space after if

File: phpseclib/Net/SSH2.php [hunk 2]
```diff
@@ -3774,7 +3774,11 @@ function getErrors()
...
+            return null;
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: just `return;` is fine

Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: or maybe, just write:

```php
$count = count($this->errors);

if($count > 0) {
    return $this->errors[$count - 1];
}
```

**Comment by splitice** [comment]
changed

**Comment by terrafrost** [comment]
Looks good! Can you squash the commits into one? I suppose I can do that too. If you don't I'll probably do so in a few days. I'm traveling atm so my availability is a bit sporadic.

**Comment by splitice** [comment]
Should be good now

**Comment by bantu** [comment]
Patch looks good to me. This is a bug fix though and will have to go into 1.0 and 2.0 too. I'll take care of that.

**Comment by bantu** [comment]
Replaced by #883

**Pull request closed without merging**